OpenShift-Sandbox-Deployment/README.md

```diff
@@ -0,0 +1,202 @@
+    keyrock-0                    1/1     Running   0          3h3m
+    mongodb-64949fb874-8fsh2     1/1     Running   0          3m41s
+    mysql-664b9568bf-sh9k2       1/1     Running   0          3h14m
+    orion-ld-58ddb4bcfb-rpx4x    1/1     Running   0          32s
+```
+
+12. Install Kong:
+
+Insert certificate and key into the [secret-file](./kong/template/secret.yaml). Make sure to base64-encode it.
+Update the configuration for the backend service(usually the context broker) to be secured in the [config-map](./kong/template/configmap.yaml). 
+
+> :warning: Do not forget the ```--skip-crds```. The sandbox does not allow cluster-wide CRDs and we dont need them in our use-case.
+
+```shell
+    helm dependency build ./kong/
+    helm install kong ./kong/ -n <YOUR_ACCOUNT>-dev --skip-crds
+```
+
+Verify:
+
+```shell
+    kubectl get pods -n <YOUR_ACCOUNT>-dev
+
+    NAME                         READY   STATUS    RESTARTS   AGE
+    keyrock-0                    1/1     Running   0          159m
+    kong-kong-64fb477547-42d2p   1/1     Running   0          35s
+    mongodb-7d4b49b5f9-96npm     1/1     Running   0          145m
+    mysql-664b9568bf-sh9k2       1/1     Running   0          170m
+```
+
+The system now can be reached at: 
+
+```shell
+   kubectl get routes kong-route -n <YOUR_ACCOUNT>-dev -o json | jq -r '.spec.host'
+```
+
+## Try out the setup
+
+Put your client key and certificate into the folder to mount(the files in  [example-secrets](./doc/example-secrets/)) are only generated examples) and use the correct IDs.
+
+1. Generate a [JWT](https://dev.ishareworks.org/introduction/jwt.html) for your client:
+```shell
+    docker run -v $(pwd)/doc/example-secrets:/certificates -e I_SHARE_CLIENT_ID="EU.EORI.NLHAPPYPETS" -e I_SHARE_IDP_ID="EU.EORI.NLPACKETDEL"  quay.io/wi_stefan/ishare-token
+```
+
+The token will be print out on the command-line. Be aware that it is(as defined by iShare) only valid for 30s.
+
+2. Request a token: 
+```shell 
+    curl --location --request POST 'https://ar.isharetest.net/connect/token' \
+        --header 'Content-Type: application/x-www-form-urlencoded' \
+        --data-urlencode 'grant_type=client_credentials' \
+        --data-urlencode 'scope=iSHARE' \
+        --data-urlencode 'client_assertion_type=urn:ietf:params:oauth:client-assertion-type:jwt-bearer' \
+        --data-urlencode 'client_assertion=<GENERATED_TOKEN>' \
+        --data-urlencode 'client_id=EU.EORI.NLHAPPYPETS'
+```
+
+The response will look  like:
+
+```json
+{
+    "access_token": "<TOKEN>",
+    "expires_in": 3600,
+    "token_type": "Bearer"
+}
+```
+
+3. Request at kong:
+
+```shell
+curl --location --request GET '<KONG-ADDRESS>/orion/ngsi-ld/v1/entities/urn:ngsi-ld:TEST:ENTITY' \
+--header 'Authorization: Bearer <ACCESS_TOKEN' 
+```
+    
+This should lead to the response:
+    
+```json
+  {
+      "message": "Local AR policy not authorized: Policy has expired or is not yet valid"
```

Comment:
This appears when there is an expired policy present at the AR for this consumer. In the case of an empty AR this will rather be:
Error when retrieving policies from local AR: <Some error message from the AR>

Reply:
Clarified